fastlane/lib/fastlane/actions/xcode_select.rb

```diff
@@ -18,32 +18,92 @@ module Actions
     #   environment variable into the full Developer content path.
     #
     class XcodeSelectAction < Action
-      def self.run(params)
-        params = nil unless params.kind_of? Array
-        xcode_path = (params || []).first
 
-        # Verify that a param was passed in
-        UI.user_error!("Path to Xcode application required (e.x. \"/Applications/Xcode.app\")") unless xcode_path.to_s.length > 0
+      def self.find_xcode(req)
+        req = Gem::Requirement.new(req.to_s)
+
+        Gem::Specification.find_by_name('xcode-install')
+        require 'xcode/install'
+
+        installer = XcodeInstall::Installer.new
+        installed = installer.installed_versions.reverse
+        installed.detect do |xcode|
+          req.satisfied_by? Gem::Version.new(xcode.version)
+        end
+      rescue Gem::LoadError
+        UI.error("The 'xcode-install' gem is needed to select Xcode by version number.")
+        Actions.verify_gem!('xcode-install') # This is awkward and destined to fail but is used to get the install help messages
```

Comment:
This was designed to be called on top of the action to tell the user how to install the gem. Any reason why we don't just add it on top of `find_xcode`?